fail to build go-json-experiment

tinygo version 0.36.0 darwin/arm64 (using go version go1.24.1 and LLVM version 19.1.2)
```
# github.com/go-json-experiment/json
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:784:8: k.SetIterKey undefined (type addressableValue has no field or method SetIterKey)
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:798:8: v.SetIterValue undefined (type addressableValue has no field or method SetIterValue)
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:806:8: k.SetIterKey undefined (type addressableValue has no field or method SetIterKey)
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:834:8: k.SetIterKey undefined (type addressableValue has no field or method SetIterKey)
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:836:8: v.SetIterValue undefined (type addressableValue has no field or method SetIterValue)
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_inlined.go:133:8: mk.SetIterKey undefined (type addressableValue has no field or method SetIterKey)
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_inlined.go:145:8: mk.SetIterKey undefined (type addressableValue has no field or method SetIterKey)
```

Tried [0.37.0-dev-e5a8d693](https://github.com/tinygo-org/tinygo/commit/e5a8d693), `SetIterKey` and `SetIterValue` methods are resolved, but still remains some issues
```
tinygo version 0.37.0-dev-e5a8d693 darwin/arm64 (using go version go1.24.1 and LLVM version 19.1.2)

# github.com/go-json-experiment/json
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:829:39: undefined: reflect.SliceOf
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:830:39: undefined: reflect.SliceOf
../../../go/pkg/mod/github.com/go-json-experiment/json@v0.0.0-20250223041408-d3c622f1b874/arshal_default.go:927:36: undefined: reflect.MapOf
```

--- a/src/reflect/type.go
+++ b/src/reflect/type.go
@@ -499,3 +499,33 @@ type StructTag = reflectlite.StructTag
 func TypeFor[T any]() Type {
 	return toType(reflectlite.TypeFor[T]())
 }
+
+func SliceOf(t Type) Type {
+	return toType(reflectlite.SliceOf(toRawType(t)))
+}
+
+func ArrayOf(n int, t Type) Type {
+	return toType(reflectlite.ArrayOf(n, toRawType(t)))
+}
+
+func StructOf([]StructField) Type {
+	return toType(reflectlite.StructOf([]reflectlite.StructField{}))
+}
+
+func MapOf(key, value Type) Type {
+	return toType(reflectlite.MapOf(toRawType(key), toRawType(value)))
+}
+
+func FuncOf(in, out []Type, variadic bool) Type {
+	rawIn := make([]reflectlite.Type, len(in))
+	for i, t := range in {
+		rawIn[i] = toRawType(t)
+	}
+
+	rawOut := make([]reflectlite.Type, len(out))
+	for i, t := range out {
+		rawOut[i] = toRawType(t)
+	}
+
+	return toType(reflectlite.FuncOf(rawIn, rawOut, variadic))
+}

